napalm/nxos/nxos.py

```diff
@@ -69,14 +70,21 @@ def __init__(self, hostname, username, password, timeout=60, optional_args=None)
         elif self.transport == 'http':
             self.port = optional_args.get('port', 80)
 
+        self.recorder_options = {
+            "mode": optional_args.get("recorder_mode", "pass"),
+            "path": optional_args.get("recorder_path", ""),
+        }
+
     def open(self):
         try:
-            self.device = NXOSDevice(self.hostname,
-                                     self.username,
-                                     self.password,
-                                     timeout=self.timeout,
-                                     port=self.port,
-                                     transport=self.transport)
+            self.device = Recorder(NXOSDevice,
+                                   recorder_options=self.recorder_options,
+                                   hostname=self.hostname,
```

Comment:
Should be `host=`, not `hostname=`
https://github.com/networktocode/pynxos/blob/48583d924b0ee171f43545c2c909b5d44aeb91e8/pynxos/device.py#L17